docs/Reek-4-to-Reek-5-migration.md

```diff
@@ -0,0 +1,180 @@
+ 
+This means that this configuration was perfectly valid:
+
+```yaml
+detectors:
+  UncommunicativeMethodName:
+    accept:
+      - !ruby/regexp /foobar/
+  UnusedPrivateMethod:
+    exclude:
+      - !ruby/regexp /i am(.*)unused/  
+```
+
+Support for this has been scrapped with Reek 5 to make the Reek configuration more yaml standard compliant.
+You can still pass in regexes, you just have to wrap them into a string using a forward slash at the
+beginning and at the end of the string like this:
+
+```Yaml
+---
+UncommunicativeMethodName:
+  accept:
+    - "/^foobar$/"
+UnusedPrivateMethod:
+  exclude:
+    - "/i am(.*)unused/"  
+```
+
+Everything within the forward slashes will be loaded as a regex.
+
+### No more single item shortcuts for list items 
+
+You cant use a configuration option that is supposed to be a list with a single element like this anymore:
+
+```Yaml
+---
+UncommunicativeMethodName:
+  accept: foobar
+UnusedPrivateMethod:
+  exclude: omg  
+```
+
+You'll have to use a proper list here like this:
+
+```Yaml
+---
+UncommunicativeMethodName:
+  accept: 
+    - foobar
+UnusedPrivateMethod:
+  exclude:
+    - omg  
+```
+
+## Failing on syntax errors in source files
+
+Previously Reek would just continue on syntax errors in source files which might have been convenient but
+not necessarily fitting for a tool that's all about code quality. With Reek 5, Reek will fail hard on
+invalid source files.
+
+### API changes
+
+This is something that will only affect very advanced users. In case you have no idea what this might be about
+you can just skip it or check out our [Developer API](docs/API.md).
+
+#### Allow only detector names in configuration hash
+
+In Reek 4 you could build your configuration like this:
+
+```ruby
+config_hash = { Reek::SmellDetectors::IrresponsibleModule => { 'enabled' => false } }
+```
+
+or like this:
+
+```ruby
+config_hash = { 'IrresponsibleModule' => { 'enabled' => false } }
+```
+
+Starting with Reek 5, the first way is not working anymore and the latter one is what you'll have to use.
+
+#### Do not accept a class as parameter for reek_of
+
+In the same vein as the change above you also can't use fully qualified detector names like this:
+
+```Ruby
+ reek_of(Reek::SmellDetectors::DuplicateMethodCall)
+ ```
+ 
+The only supported way now is either as symbol or string:
+ 
+```Ruby
+reek_of(:DuplicateMethodCall)
+reek_of('DuplicateMethodCall')
+```
+
+## Smaller changes
+
+* `PrimaDonnaMethod` has been given the better name `MissingSafeMethod`
+* `wiki-links` flag has been renamed to `documentation` flag
+* Reek assumes the default configuration file to be named ".reek.yml" and will ignore all other files. You can
+still use any name you want though by passing in a name via the `-c` flag
+* We have dropped the legacy code comment separator
+* We have dropped support for Ruby 2.1 and 2.2 since Ruby 2.2. is officially not supported by the Ruby core team anymore
```

Comment:
Let's make this "since they are officially not supported by the Ruby core team anymore"

Reply:
Updated.